tests/unit/scanner/test_upload_rules_db.py

```diff
@@ -45,27 +46,25 @@ def _create_module_and_rules(
     return module_yaml
 
 
-@patch("boostsec.registry_validator.upload_rules_db.requests")
-def test_upload_rules_db(mock_requests: Any, tmp_path: Path) -> None:
+def test_upload_rules_db(tmp_path: Path, requests_mock: Mocker) -> None:
     """Test upload_rules_db."""
-    mock_requests.post.return_value.status_code = 200
-    mock_requests.post.return_value.text = "RuleSuccessSchema"
+    url = "https://my_endpoint/"
+    requests_mock.post(
+        urljoin(url, "/rules-management/graphql"),
```

Comment:
I think you can assert the auth `"ApiKey my-token"` with an `additional_matcher`. check in our code base there's quite a few of them.

This can only be tested once in 1 test only, no need to pollute all the other ones.

It's possible it does not appear in the headers of the request, if that's the case you would need WireMock to test it.